packages/edit-post/src/components/visual-editor/index.js

```diff
@@ -137,16 +138,21 @@ export default function VisualEditor( { styles } ) {
 		}
 
 		const supportsTemplateMode = getEditorSettings().supportsTemplateMode;
+		const canEditTemplate = select( coreStore ).canUser(
+			'create',
+			'templates'
+		);
 
 		return {
 			deviceType: __experimentalGetPreviewDeviceType(),
 			isWelcomeGuideVisible: isFeatureActive( 'welcomeGuide' ),
 			isTemplateMode: _isTemplateMode,
 			// Post template fetch returns a 404 on classic themes, which
 			// messes with e2e tests, so we check it's a block theme first.
-			editedPostTemplate: supportsTemplateMode
-				? getEditedPostTemplate()
-				: {},
+			editedPostTemplate:
+				supportsTemplateMode && canEditTemplate
+					? getEditedPostTemplate()
+					: undefined,
```

Comment:
We should avoid returning new object references here. It will fail shallow equality check and cause unnecessary rerenders.

Reply:
It's also being given a default on line 119 where we destructure the return value of `useSelect` so I suppose we should remove that one too.

Reply:
Assigning default values during destructuring is okay; that doesn't change values returned inside the select.